platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java

```diff
@@ -308,6 +271,14 @@ public void visitDependency(Dependency dependency) {
             additionalDependencies.add(dependency);
         }
 
+        @Override
+        public void visitReplacement(ModuleIdentifier original, ModuleIdentifier replacement) {
```

Comment:
🤔 This type isn't a Visitor and doesn't do anything but accumulate data.

Maybe instead of `PluginDependencyVisitor` it should be called `DefaultPluginResolveContext` and all the `visitXYZ` methods called `recordXYZ` or something instead?